### ISSUE TITLE
Poor results with only 5 images for training – guidance to reproduce paper performance

### ISSUE THREAD

**Issue by hwzjho** [biplabpoudel25/CryoFSL#2]
Hello,

First of all, thank you very much for your excellent work and for sharing this project. I followed the instructions in the README.md and trained the model using only 5 images. However, the results are not satisfactory compared to what is reported in the paper. Here is a screenshot of my results:

<img width="132" height="375" alt="Image" src="https://github.com/user-attachments/assets/929292a4-72b0-472e-9879-103efe8c60ac" /> 
<img width="688" height="669" alt="Image" src="https://github.com/user-attachments/assets/360cc4d7-8e95-46c4-b4b7-c57104defb6d" /> 
<img width="378" height="798" alt="Image" src="https://github.com/user-attachments/assets/e36112ac-3675-46b6-8c42-c36a7158c8d6" />

Could you please provide some guidance on how to improve the performance to achieve results closer to the paper?
Any suggestions would be greatly appreciated. Thank you in advance for your help!

Best regards!

**Comment by biplabpoudel25** [comment]
Thanks for trying CryoFSL.

Please make sure:
      1. The images and masks are correctly aligned, some datasets with .mrc and .csv files may have flipped masks.
      2. The images are normalized before training/testing (ours were).
      3. You start from the pretrained SAM2 checkpoint, it’s essential for good few-shot results.

**Comment by biplabpoudel25** [comment]
I have uploaded the **training and testing** code to reproduce the result step by step. Please follow the notebook.

**Comment by hwzjho** [comment]
Thank you so very much. Let me give it a try.

**Comment by hwzjho** [comment]
Thank you very much again! I can now run it successfully. Wish you all the best!
